saw-central/src/SAWCentral/Crucible/MIR/MethodSpecIR.hs

```diff
@@ -115,6 +124,18 @@ mutIso =
 isMut :: Lens' (MirAllocSpec tp) Bool
 isMut = maMutbl . mutIso
 
+-- | Converts a 'MirPointerKind' to a 'M.Ty' constructor.
+ptrKindToTy :: MirPointerKind -> M.Ty -> M.Mutability -> M.Ty
+ptrKindToTy MirPointerRef = M.TyRef
+ptrKindToTy MirPointerRaw = M.TyRawPtr
+
+-- | Gets the 'MirPointerKind' for a reference or pointer type.
+tyToPtrKind :: M.Ty -> MirPointerKind
+tyToPtrKind (M.TyRef _ _) = MirPointerRef
+tyToPtrKind (M.TyRawPtr _ _) = MirPointerRaw
+tyToPtrKind ty = panic "tyToPtrKind"
+  ["Type is not a reference or pointer: " <> Text.pack (show ty)]
```

Comment:
I'm not really sure where to put these so I put them in this module. Let me know if there's a better place to put them.